exporter/datadogexporter/factory.go

```diff
@@ -237,7 +239,7 @@ func checkAndCastConfig(c component.Config, logger *zap.Logger) *Config {
 	return cfg
 }
 
-func (f *factory) consumeStatsPayload(ctx context.Context, out chan []byte, traceagent *agent.Agent, tracerVersion string, logger *zap.Logger) {
+func (f *factory) consumeStatsPayload(ctx context.Context, statsIn <-chan []byte, statsToAgent chan<- *pb.StatsPayload, logger *zap.Logger) {
```

Comment:
This only get's called when the `ConnectorPerformanceFeatureGate` is enabled. This is going to be the default at some point, but for now it isn't and I believe we also need make the change [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2bf56cacf1428e13483f2dea5ef237475f3ca48f/exporter/datadogexporter/metrics_exporter.go#L263-L265)